### PR TITLE
Reorganized Store class. Rewrite classes for polymorphism. Added buy cargo features

### PR DESCRIPTION
**Pull request by mateusz-talma**

### CODE REVIEW

File: shm/player.cpp [hunk 1]
```diff
@@ -53,34 +41,17 @@ void Player::nextDay(){
...
-}
-
-void  Player::removeFruit(Fruit* cargo) {
-    ship_->removeFruit(cargo);
-}
-
-void  Player::removeItem(Item* cargo) {
-    ship_->removeItem(cargo);
+size_t Player::getAvailableSpace() { 
+    return availableSpace_ = ship_->countAvailableSpace();
```
Comment (WBigaj) [inline shm/player.cpp:45]: Do we need availableSpace_ in the player class. Can't we just return ship_->countAvailableSpace()?

**Review by WBigaj — APPROVED** [review]
I have one question.

File: shm/player.cpp (hunk 1 above)
Comment (mateusz-talma) [reply shm/player.cpp:45]: Yes, you're right. Maybe this field will be useful in the future, when the player will have more ships :)

**Review by Marcinati — APPROVED** [review]
well done

**Pull request merged**